### PR TITLE
Added support for encoding 8-bit grayscale images (colorType = 0 and 4)

### PR DESCRIPTION
**Pull request by ekulabuhov**
In connection to #75.

Haven't added unit tests or README changes yet. Could someone review the changes, please?

### CODE REVIEW

File: lib/bitpacker.js [hunk 1]
```diff
@@ -3,15 +3,15 @@
 var constants = require('./constants');
 
 module.exports = function(data, width, height, options) {
-  var outHasAlpha = options.colorType === constants.COLORTYPE_COLOR_ALPHA;
-  if (options.inputHasAlpha && outHasAlpha) {
+  var outHasAlpha = options.colorType === constants.COLORTYPE_COLOR_ALPHA || constants.COLORTYPE_ALPHA;
```
Comment (lukeapage) [inline lib/bitpacker.js]: won't this make outHasAlpha always truthy?

File: lib/bitpacker.js [hunk 2]
```diff
@@ -48,11 +48,22 @@ module.exports = function(data, width, height, options) {
...
+          outData[outIndex + 2] = blue;
+          if (outHasAlpha) {
+            outData[outIndex + 3] = alpha;
+          }
+          break;
+        case constants.COLORTYPE_ALPHA:
+          // Convert to grayscale and alpha
+          var grayscale = (red + green + blue) / 3;
+          outData[outIndex] = grayscale;
+          outData[outIndex + 1] = alpha;
```
Comment (lukeapage) [inline lib/bitpacker.js]: you don't need an if here for outHasAlpha too ?

File: lib/bitpacker.js (hunk 1 above)
Comment (ekulabuhov) [inline lib/bitpacker.js]: You're right, this line doesn't make sense. Not sure what I was thinking there.

**Comment by ekulabuhov** [comment]
Fixed failing unit-tests. Updated README. 

Now, I'm not sure how to setup a unit test for encoder? Could you help me with that?

**Comment by patrickhulce** [comment]
This is great, is there anything I can do to help here?

**Comment by lukeapage** [comment]
Thanks for this, released in 3.2.0

**Pull request merged**